### PR TITLE
feature: custom media export option

### PR DESCRIPTION
**Pull request by dnnsjrng**
Add a simple option to export custom-media according to the W3C draft https://www.w3.org/TR/mediaqueries-5/#at-ruledef-custom-media

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,6 +96,18 @@ theme_name.lg:
...
+@custom-media --Themename-small-mediaQuery (only screen and (max-width: 35.5625rem));
+@custom-media --Themename-small-resolution (2x);
+@custom-media --Themename-medium-mediaQuery (only screen and (min-width: 35.625rem));
+@custom-media --Themename-medium-resolution (2x);
+@custom-media --Themename-medium-minWidth (35.625rem);
+@custom-media --Themename-medium-maxWidth (35.625rem);
+@custom-media --Themename-large-mediaQuery (only screen and (min-width: 48rem) and (max-width: 63.9375rem));
+@custom-media --Themename-large-resolution (2x);
+@custom-media --Themename-large-minWidth (48rem);
+@custom-media --Themename-large-maxWidth (48rem);
```
Comment (LucidityDesign) [inline README.md:108]: Min and max shouldn't overlap

```suggestion
@custom-media --Themename-large-minWidth (48rem);
@custom-media --Themename-large-maxWidth (47.9375rem);
```

File: schema/drupal-breakpoints-css.json [hunk 2]
```diff
@@ -72,6 +72,9 @@
         },
         "element": {
           "type": "string"
+        },
+        "customMedia": {
+          "type": "string"
```
Comment (mvsde) [inline schema/drupal-breakpoints-css.json:77]: I think this should be a `boolean`.

File: lib/index.js [hunk 3]
```diff
@@ -218,19 +218,30 @@ module.exports = class Breakpoints {
...
+        // Ref: https://www.w3.org/TR/mediaqueries-5/#at-ruledef-custom-media
+        string += `@custom-media --${customProperty.identifier} (${customProperty.value});`;
+      });
+    } else {
+      string += `${this.#config.css.element} {`;
+      this.#customProperties.forEach((customProperty) => {
+        string += `--${customProperty.identifier}: "${customProperty.value}";`;
+      });
+      string += `}`;
+    }
```
Comment (mvsde) [inline lib/index.js:241]: I wonder if custom media queries and custom properties should be exclusionary. Maybe someone needs both?

**Review by mvsde — CHANGES_REQUESTED** [review]

**Review by mvsde — APPROVED** [review]
Looks good to me 👍

**Pull request merged**